datadog exporter panic

### Component(s)

exporter/datadog

### What happened?

## Description

datadog exporter panics

## Steps to Reproduce

:shrug: run the exporter, connected to vector

## Expected Result

no panic

## Actual Result

panic, see log

### Collector version

0.70.0

### Environment information

otelcol-contrib docker image

### OpenTelemetry Collector configuration

_No response_

### Log output

```shell
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x25591b3]

goroutine 247 [running]:
github.com/open-telemetry/opentelemetry-collector-contrib/exporter/datadogexporter/internal/clientutil.isNonRetriable(...)
	github.com/open-telemetry/opentelemetry-collector-contrib/exporter/datadogexporter@v0.70.0/internal/clientutil/error_converter.go:32
github.com/open-telemetry/opentelemetry-collector-contrib/exporter/datadogexporter/internal/clientutil.WrapError(...)
	github.com/open-telemetry/opentelemetry-collector-contrib/exporter/datadogexporter@v0.70.0/internal/clientutil/error_converter.go:25
github.com/open-telemetry/opentelemetry-collector-contrib/exporter/datadogexporter.(*metricsExporter).pushSketches(0xc0016e6500, {0x780ef70, 0xc08aa40000}, {0xc0bfb03000?, 0xc0367d9098?, 0xcfe88a?})
	github.com/open-telemetry/opentelemetry-collector-contrib/exporter/datadogexporter@v0.70.0/metrics_exporter.go:171 +0x353
github.com/open-telemetry/opentelemetry-collector-contrib/exporter/datadogexporter.(*metricsExporter).PushMetricsData.func4({0x780ef70?, 0xc08aa40000?})
	github.com/open-telemetry/opentelemetry-collector-contrib/exporter/datadogexporter@v0.70.0/metrics_exporter.go:251 +0x3c
github.com/open-telemetry/opentelemetry-collector-contrib/exporter/datadogexporter/internal/clientutil.(*Retrier).DoWithRetries(0xc0016d5840, {0x780ef70, 0xc08aa40000}, 0xc0367d9500)
	github.com/open-telemetry/opentelemetry-collector-contrib/exporter/datadogexporter@v0.70.0/internal/clientutil/retrier.go:65 +0x189
github.com/open-telemetry/opentelemetry-collector-contrib/exporter/datadogexporter.(*metricsExporter).PushMetricsData(0xc0016e6500, {0x780ef70, 0xc08aa40000}, {0xc0367d9868?})
	github.com/open-telemetry/opentelemetry-collector-contrib/exporter/datadogexporter@v0.70.0/metrics_exporter.go:250 +0xc0c
github.com/open-telemetry/opentelemetry-collector-contrib/exporter/datadogexporter.(*metricsExporter).PushMetricsDataScrubbed(0xc0016e6500, {0x780ef70?, 0xc08aa40000?}, {0x0?})
	github.com/open-telemetry/opentelemetry-collector-contrib/exporter/datadogexporter@v0.70.0/metrics_exporter.go:182 +0x28
go.opentelemetry.io/collector/exporter/exporterhelper.(*metricsRequest).Export(0x424607?, {0x780ef70?, 0xc08aa40000?})
	go.opentelemetry.io/collector@v0.70.0/exporter/exporterhelper/metrics.go:65 +0x34
go.opentelemetry.io/collector/exporter/exporterhelper.(*timeoutSender).send(0xc0016b2468, {0x782ce50, 0xc090bb7b90})
	go.opentelemetry.io/collector@v0.70.0/exporter/exporterhelper/common.go:208 +0x96
go.opentelemetry.io/collector/exporter/exporterhelper.(*retrySender).send(0xc0016e25a0, {0x782ce50?, 0xc090bb7b90?})
	go.opentelemetry.io/collector@v0.70.0/exporter/exporterhelper/queued_retry.go:358 +0x185
go.opentelemetry.io/collector/exporter/exporterhelper.(*metricsSenderWithObservability).send(0xc0016d6210, {0x782ce50, 0xc090bb7b90})
	go.opentelemetry.io/collector@v0.70.0/exporter/exporterhelper/metrics.go:136 +0x88
go.opentelemetry.io/collector/exporter/exporterhelper.(*queuedRetrySender).start.func1({0x782ce50, 0xc090bb7b90})
	go.opentelemetry.io/collector@v0.70.0/exporter/exporterhelper/queued_retry.go:205 +0x39
go.opentelemetry.io/collector/exporter/exporterhelper/internal.(*boundedMemoryQueue).StartConsumers.func1()
	go.opentelemetry.io/collector@v0.70.0/exporter/exporterhelper/internal/bounded_memory_queue.go:61 +0xb6
created by go.opentelemetry.io/collector/exporter/exporterhelper/internal.(*boundedMemoryQueue).StartConsumers
	go.opentelemetry.io/collector@v0.70.0/exporter/exporterhelper/internal/bounded_memory_queue.go:56 +0x45
```
```


### Additional context

This was observed on multiple pods after upgrading from 0.66 -> 0.70

--- a/.chloggen/datadogexporter-fix-nil-dereference.yaml
+++ b/.chloggen/datadogexporter-fix-nil-dereference.yaml
@@ -1,0 +1,16 @@
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: bug_fix
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: datadogexporter
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Fix a nil dereferencing bug on http response
+
+# One or more tracking issues related to the change
+issues: [18099]
+
+# (Optional) One or more lines of additional information to render under the primary note.
+# These lines will be padded with 2 spaces and then inserted directly into the document.
+# Use pipe (|) for multiline entries.
+subtext:

--- a/exporter/datadogexporter/internal/clientutil/error_converter.go
+++ b/exporter/datadogexporter/internal/clientutil/error_converter.go
@@ -22,7 +22,7 @@ import (
 
 // WrapError wraps an error to a permanent consumer error that won't be retried if the http response code is non-retriable.
 func WrapError(err error, resp *http.Response) error {
-	if err == nil || !isNonRetriable(resp) {
+	if err == nil || resp == nil || !isNonRetriable(resp) {
 		return err
 	}
 	return consumererror.NewPermanent(err)

--- a/exporter/datadogexporter/internal/clientutil/error_converter_test.go
+++ b/exporter/datadogexporter/internal/clientutil/error_converter_test.go
@@ -32,4 +32,5 @@ func TestWrapError(t *testing.T) {
 	assert.False(t, consumererror.IsPermanent(WrapError(err, &respRetriable)))
 	assert.True(t, consumererror.IsPermanent(WrapError(err, &respNonRetriable)))
 	assert.False(t, consumererror.IsPermanent(WrapError(nil, &respNonRetriable)))
+	assert.False(t, consumererror.IsPermanent(WrapError(err, nil)))
 }

